### ISSUE TITLE
`jax.scipy.special.digamma(0.0)` returns `nan` — should be `-inf`

### ISSUE THREAD

**Issue by JamesDLance** [jax-ml/jax#37763]
### Description

`jax.scipy.special.digamma` returns `nan` at `x = 0`. The correct value is `-inf` because digamma has a simple pole at the origin. SciPy handles this correctly.

This is a boundary-value correctness bug, not a precision issue. The digamma function is well-defined near zero (it diverges to −∞), and any implementation should return `-inf` rather than `nan` for `x = 0`.

I ran into this while computing gradients of a Dirichlet log-likelihood. The gradient involves `digamma(alpha_k)` and when one concentration parameter is very small, the result should be a large negative number — not `nan` which poisons the entire gradient vector.

**Secondary issue**: because XLA flushes denormal inputs to zero (DAZ), subnormal float32 inputs also trigger the `x = 0` code path, so `digamma(subnormal)` returns `nan` too. The correct value for small positive `x` is approximately `−1/x`, which would be a large negative number or `-inf` for the smallest subnormals.

## Reproducer

```python
from jax.scipy.special import digamma as jax_digamma
from scipy.special import digamma as sp_digamma
import jax.numpy as jnp
import numpy as np

test_vals = [0.0, np.nextafter(np.finfo(np.float32).tiny, 0.0), 1e-40, 0.5, 1.0]

for v in test_vals:
    sp  = sp_digamma(v)
    jax = float(jax_digamma(jnp.float32(v)))
    print(f"digamma({v:.2e}):  scipy={sp:.4g}   jax={jax}")
```

## Expected output

```
digamma(0.00e+00):  scipy=-inf        jax=-inf
digamma(1.18e-38):  scipy=-8.507e+37  jax=-8.507e+37
digamma(1.00e-40):  scipy=-1e+40      jax=-inf
digamma(5.00e-01):  scipy=-1.964      jax=-1.964
digamma(1.00e+00):  scipy=-0.5772     jax=-0.5772
```

## Actual output

```
digamma(0.00e+00):  scipy=-inf        jax=nan    ← wrong
digamma(1.18e-38):  scipy=-8.507e+37  jax=nan    ← wrong
digamma(1.00e-40):  scipy=-1e+40      jax=nan    ← wrong
digamma(5.00e-01):  scipy=-1.964      jax=-1.964 ← ok
digamma(1.00e+00):  scipy=-0.5772     jax=-0.5772 ← ok
```

## Notes

- The `x = 0` case is the primary bug. The subnormal failures are a consequence of it combined with XLA's DAZ mode.
- `jnp.log(0.0)` correctly returns `-inf`, so the issue is specific to digamma's boundary handling, not a general `-inf` problem in JAX.
- Normal-range inputs work correctly.

### System info (python version, jaxlib version, accelerator, etc.)

jax:     0.10.1.dev20260513+64f6e229b
jaxlib:  0.10.0
numpy:   2.4.4
scipy:   1.17.1
python:  3.12.3
OS:      Linux-6.14.0-37-generic-x86_64-with-glibc2.39
backend: cpu

**Comment by jakevdp** [comment]
Here's the StableHLO code where the `NaN` comes from: https://github.com/openxla/stablehlo/blob/effc2e60751b8e900f11c612c3f5b07e2a5bbfd4/stablehlo/transforms/ChloLegalizeToStablehlo.cpp#L1635

**Comment by jakevdp** [comment]
Note the comment on this issue here: https://github.com/jax-ml/jax/pull/37917#issuecomment-4556664580

The current behavior is arguably correct.

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
